AsyncDisplayKit/Details/ASTextNodeRenderer.mm

```diff
@@ -173,9 +173,13 @@ - (void)_calculateSize
 {
   ASDN::MutexLocker l(_textKitLock);
 
+  if (_attributedString.string.length == 0) {
```

Comment:
Small change, but could you make sure this checks _attributedString.length directly?

I'm not completely sure if the .string property always has an NSString allocated & ready (it might), but because this is an essential codepath that every text label in ASDK apps will hit, it's worth being cautious and minimal.

Thanks for removing the extraneous space :)